subsys/bluetooth/host/att.c

```diff
@@ -3868,8 +3868,11 @@ int bt_att_req_send(struct bt_conn *conn, struct bt_att_req *req)
 		return -ENOTCONN;
 	}
 
+	/* att_req_send_process is not thread-safe */
+	k_sched_lock();
```

Comment:
I think we should include the above ENOTCONN check inside the scheduler lock. It also looks thread unsafe.

Reply:
To be more clear: The `sys_slist_append` below should be atomic with the test above. Otherwise we risk appending to a disconnected att.

Reply:
thanks, I added that reason to the commit message too.